h2/src/main/org/h2/command/ddl/AlterType.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2004-2024 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (https://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.command.ddl;
+
+import org.h2.api.ErrorCode;
+import org.h2.command.CommandInterface;
+import org.h2.engine.SessionLocal;
+import org.h2.message.DbException;
+import org.h2.schema.Domain;
+import org.h2.schema.Schema;
+import org.h2.value.ExtTypeInfoEnum;
+import org.h2.value.TypeInfo;
+import org.h2.value.Value;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class represents the statements
+ * ALTER TYPE
+ */
+public class AlterType extends SchemaCommand {
+
+    private String typeName;
+    private String value;
+
+    public AlterType(SessionLocal session, Schema schema) {
+        super(session, schema);
+    }
+
+    public void setTypeName(String typeName) {
+        this.typeName = typeName;
+    }
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public long update() {
+        Domain domain =  getSchema().findDomain(typeName);
+        if (domain == null) {
+            throw DbException.get(ErrorCode.DOMAIN_NOT_FOUND_1, typeName);
+        }
+
+        TypeInfo dataType = domain.getDataType();
+        if (dataType.getValueType() != Value.ENUM) {
+            throw DbException.get(ErrorCode.WRONG_OBJECT_TYPE, typeName);
+        }
+
+        ExtTypeInfoEnum oldExtTypeInfo = (ExtTypeInfoEnum) dataType.getExtTypeInfo();
+        int count = oldExtTypeInfo.getCount();
+        List<String> newValues = new ArrayList<>();
+        for (int i = 0; i < count; i++) {
+            newValues.add(oldExtTypeInfo.getEnumerator(i));
+        }
+        newValues.add(value);
+
+        domain.setDataType( TypeInfo.getTypeInfo(
+                Value.ENUM,
+                -1L,
+                -1,
+                new ExtTypeInfoEnum(newValues.toArray(new String[0]))));
```

Comment:
Please, format your code with the same code style as project uses.

Reply:
Fixed